primary/src/core.rs

```diff
@@ -263,38 +265,77 @@ impl Core {
             .inc();
 
         // Check if we can vote for this header.
-        // TODO: compute from a persisted last_voted (issue #488)
-        if self
-            .last_voted
-            .entry(header.round)
-            .or_insert_with(HashSet::new)
-            .insert(header.author.clone())
-        {
-            // Make a vote and send it to the header's creator.
-            let vote = Vote::new(header, &self.name, &mut self.signature_service).await;
-            debug!(
-                "Created vote {vote:?} for {header} at round {}",
-                header.round
-            );
-            if vote.origin == self.name {
-                // This used to be a panic, but this may fail if we're processing a late-received
-                // header where we were already are part of the signers (as a restart could have blown our `last_voted`)
-                // TODO: change this back to a panic-on-error once we have persisted last_voted (issue #488)
-                let res = self.process_vote(vote).await;
-                if let Err(e) = res {
-                    error!("Failed to process our own vote: {}", e.to_string());
+        // Send the vote when:
```

Comment:
I believe that the `process_header` method started becoming quite big and having increased responsibility. Do you see any parts that could be extracted in a method to start reducing its size a bit? Nothing blocking here, but since we are touching this part it might be beneficial

Reply:
👍 extracting the send_vote portion